components/remote_settings/src/client.rs

```diff
@@ -29,6 +51,7 @@ pub struct RemoteSettingsClient<C = ViaductApiClient> {
     collection_name: String,
     jexl_filter: JexlFilter,
     inner: Mutex<RemoteSettingsClientInner<C>>,
+    packaged_data: PackagedData,
```

Comment:
Is this going to mean that we have multiple copies (or even a single copy) of the full packaged data in memory for the lifetime of the app?

Given the `defaults.json` file is 1Mb, I'm concerned about the overhead here. On desktop, if the cache is either empty, invalid or out of date, then we'll read the dump files, otherwise, we skip loading the dumps completely.

Reply:
I have a patch ready to split the files up. But I don't think we have anything in place to if/else load the attachments. We currently had in mind to always package them, and then during runtime decide to read from them.

Reply:
I think the main difference is that as a component, we don't have the ability to package files and load them at runtime.  So the solution here is to load the file at compile time and store it in the binary.

One alternative would be that the application hands us the packaged data in some way.  Maybe Desktop could pass in the directory of the dumps.  This would allow us to determine at runtime what we want to load.  This seems a lot more complicated to me though.

A micro optimization could be to store the data in an `Arc` to limit the number of copies to 1.  I think this won't matter so much when the data is split up since each client will only get the data for their collection, but it's still possible to create 2 clients for the same collection.

Reply:
Where would we store these JSON files then? If we want to read them on demand, we need to have access to them. Which means we need a cross-platform way of shipping and storing these files. 

Do we do this for any other component before?

Reply:
We've never stored files on our own before, I don't think that's something we should be doing in this PR.

IIUC, one suggestion from @Standard8 would be to still use `include_str!` for the JSON source, but not store the parsed JSON objects in memory.  That way we have one less copy of the data in RAM.  That seems reasonable to me, since you store the data in the database when you realize the packaged data is newer.

Reply:
Now we need to know which `collections` we need and want, and from which `bucket`. Can someone shed a light on this?